base_classes/NXparameters.nxdl.xml

```diff
@@ -26,11 +26,69 @@
     xsi:schemaLocation="http://definition.nexusformat.org/nxdl/3.1 ../nxdl.xsd"
     name="NXparameters" 
 	type="group" extends="NXobject">
-    <doc>Container for parameters, usually used in processing or analysis.</doc>
+    <doc>
+        Container for parameters used in processing or analysing data.
+
+        This base class can be used in a number of different contexts,
+        and is allowed in any other base class. However, a common use
+        case is to store the results of optimizing parameters, for
+        example, by least-squares fitting, so a number of attributes
+        have been defined for this purpose. Normally, this group will be
+        stored in a :ref:NXprocess group, which defines the program used
+        to process these parameters. If multiple functions are used to
+        analyze the data, there can be additional NXparameter groups for
+        each one.        
+    </doc>
+    <attribute name="model">
+        <doc>
+            The name of the model used in optimizing the data. Fitting
+            packages such as LMFIT (https://lmfit.github.io/lmfit-py/)
+            provide models, which instantiate functions to be fitted to
+            the data. If this attribute is provided, it is assumed that
+            all the parameters in this group are associated with this
+            model.
+        </doc>
+    </attribute>
     <field name="TERM" minOccurs="0" maxOccurs="unbounded" nameType="any">
-        <!-- maxOccurs="unbounded" is intended but is not allowed by current syntax -->
-        <doc>A parameter (also known as a term) that is used in or results from processing.</doc>
-        <attribute name="units"/>
+        <doc>
+            A parameter (also known as a term) that is used in or
+            results from processing. 
+        </doc>
+        <attribute name="units" />
+        <attribute name="error" type="NX_NUMBER">
+            <doc>
+                The standard deviation resulting from data optimization.
+            </doc>
+        </attribute>
+        <attribute name="expression">
+            <doc>
+                A string representing an expression that can be used to
+                relate the parameter to another parameter's value. The
+                format of this string is dependent on the program used
+                to optimize the parameters and is not specified by
+                NeXus.
+            </doc>
+        </attribute>
+        <attribute name="initial_value" type="NX_NUMBER">
+            <doc>
+                The initial value of the parameter.
+            </doc>
+        </attribute>
+        <attribute name="max" type="NX_NUMBER">
+            <doc>
+                The upper bound of the parameter value.
+            </doc>
+        </attribute>
+        <attribute name="min" type="NX_NUMBER">
```

Comment:
Rather suggestion: min and max I agree that these are  useful as summary statistics for especially quantities would it be an (additional idea) to make _min, _max reserved suffixes that would cover for this?

Reply:
If you are suggesting that these should be fields, I would prefer they remain as attributes for the reasons I gave [above](https://github.com/nexusformat/definitions/pull/1560#discussion_r2071684581).